### PR TITLE
Validate layer indexes during command parsing

### LINKED ISSUE #316: Validate layer indexes during command parsing
## Problem

Layer-index help does not consistently state that indexes are zero-based. Negative values are accepted by the parser, and range validation occurs only after the image manifest has been fetched, causing avoidable registry work before reporting a user-input error.

## Proposed improvement

Describe every layer-index option as zero-based and reject negative values during command parsing. Manifest-dependent upper-bound checks can remain after manifest retrieval, with errors that include the valid range.

Add parser tests for negative values and command tests for indexes at and beyond the manifest boundary.

> This issue was generated by GitHub Copilot from an AI-assisted CLI API and UX review.

### PR DESCRIPTION
**Pull request by mthalman**
Negative layer indexes were accepted until after the image manifest had been fetched, causing avoidable registry work and producing vague range errors.

This change centralizes layer-index option construction so all affected options advertise zero-based indexing and reject negative values during parsing. Manifest-dependent upper-bound validation remains in `ImageHelper`, with explicit valid ranges and a dedicated message for images with no layers. The comparison commands now share their full option names between registration and runtime diagnostics.

Tests cover negative, malformed, overflowing, culture-specific, boundary, and empty-manifest inputs across .NET 9 and .NET 10. The complete Docker-backed suite passes with 501 tests per target framework.

Fixes: #316

### CODE REVIEW

File: src/Valleysoft.Dredge/Commands/Image/LayerIndexOption.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                    out int value) &&
+                value < 0)
+            {
+                result.AddError($"Layer index for option '{name}' must be zero or greater.");
+            }
+        });
+
+        return option;
+    }
+}
```
[the remark this answers is not in the document]
Comment (mthalman) [reply src/Valleysoft.Dredge/Commands/Image/LayerIndexOption.cs:30]: Using GetValueOrDefault<int?>() here causes System.CommandLine 2.0.11 to throw InvalidOperationException when conversion has already failed, which crashes the CLI for non-numeric and overflowing values. Parsing the single raw token with the same NumberStyles.Integer and CurrentCulture semantics avoids that regression while leaving malformed values to the framework's normal conversion error. The behavior is covered by malformed, overflow, negative, and culture-specific parser tests.

_Auto-resolved by the [GitHub Copilot app](https://github.com/github/app)._
